### PR TITLE
[WIP] Test for merge function

### PR DESCRIPTION
**Pull request by peterclemenko**
This is a test and RFC for merge functionality. Do not merge this yet, as it is untested.

### CODE REVIEW

File: __test__/relation-tests.js [hunk 1]
```diff
@@ -44,6 +44,30 @@ test('Test CREATE relation', (assert) => {
...
 });
 
+
+
+test('Test MERGE relation', (assert) => {
+    let node1 = nodes[0];
+    let node2 = nodes[1];
+    let rel1 = OGMNeoRelation.relateMerge(node1.id, 'relatedto', node2.id, { property: 'a' });
+    let rel2 = OGMNeoRelation.relateMerge(node1.id, 'relatedto', node2.id, {});
+    let rel3 = OGMNeoRelation.relateMerge(node1.id, 'relatedto', node2.id, { property: 'a' });
```
Comment (LucianoPAlmeida) [inline __test__/relation-tests.js]: rel3 and rel4 aren't being used here on this test

**Review by LucianoPAlmeida — COMMENTED** [review]
Hey @aoighost I think it's good, post some comments above. Just some linting problems and some tests that actually test node counts that fail because they are being affected by the new tests that create a new node.

File: __test__/relation-tests.js (hunk 1 above)
Comment (peterclemenko) [reply __test__/relation-tests.js]: Alright I forgot about this. I'll fix it shortly.

**Comment by peterclemenko** [comment]
Alright, I forgot about this. I'm probably going to add create unique as well in an updated version of this, hang tight.

**Comment by peterclemenko** [comment]
Reading up again, create unique is deprecated. Never mind, I'll just go with merge.

**Comment by peterclemenko** [comment]
OK, from the looks of it my code is fixed and hopefully should be mergeable @LucianoPAlmeida. That being said, looks like travis is finding test issues in the base code, so you might want to take a look.

**Comment by LucianoPAlmeida** [comment]
Hey @aoighost 
Thank's for that :)) 
I think the Travis error is just in some tests that return the nodes counts and since you create new nodes on the merge tests it was expecting like one and got two. There are a few solutions to that:
* Update tests 'Test updateMany node' and 'Test delete MANY NODE' expectations to + 1.
* Delete the nodes created on the test merge after the test completes.
* Or just use another label for the nodes on the merge tests that you implemented. 

That's actually not the good scenario for unit tests to impact each other, my bad on that. Can you make this changes? Any of those are acceptable, I'll probably decouple the unit tests later so they don't depend on each other.
After that, I think it will be good to go :))

**Pull request merged**